### ISSUE TITLE
Method afterStateUpdated() is called twice

### ISSUE THREAD

**Issue by vuolter** [laravel-filament/filament#6576]
### Package

filament/forms

### Package Version

v2.17.40

### Laravel Version

v10

### PHP Version

PHP 8.2.0

### Problem description

Method `afterStateUpdated` is called twice as already explained in discussion https://github.com/filamentphp/filament/discussions/5351.

### Expected behavior

Should be called only one time after the field value is changed.

### Steps to reproduce

Just add `->reactive` and `->afterStateUpdated` to any form field and log the calls count.

### Reproduction repository

https://github.com/filamentphp/demo

**Comment by vuolter** [comment]
Are you kidding bot? LOL

@danharrin sorry to bother you, but I guess there is something wrong here...

**Comment by danharrin** [comment]
@vuolter the bot is not wrong, we do expect you to make your own *minimal* reproduction repository instead of using our demo. Here is why:

I've just tried to reproduce your problem myself and I can't. Maybe it is due to lack of detail in the issue. I will need an actual reproduction repository to continue. Let me know when you've made that.

**Comment by hxnk** [comment]
@danharrin I can confirm this does indeed happen when creating a Page (child of `Filament\Pages\Page`), and utilizing the `InteractsWithTable` trait on that child page.

- `InteractsWithTable` utilizes `InteractsWithForms`
- The parent `Filament\Pages\Page` utilizes `InteractsWithForms` as well (indirectly through `HasActions`)

The issue is with the parent call to `callBeforeAndAfterSyncHooks` in the `InteractsWithForms` trait here:

https://github.com/filamentphp/filament/blob/020bf2c6187317f2fe3de1fdddf80b365b9266c9/packages/forms/src/Concerns/InteractsWithForms.php#L205C3-L212

It's effectively running twice: first from the custom page context, and then from the parent page context...

**Comment by danharrin** [comment]
@hxnk please create a full reproduction repository and I will reopen the issue

**Comment by hxnk** [comment]
@danharrin I will when I find the time, I hoped the explanation would be straight forward enough... 😁

@vuolter: a workaround until this is fixed is to override `callBeforeAndAfterSyncHooks` (when utilizing `InteractsWithTable`) like this:

```php

    protected function callBeforeAndAfterSyncHooks($name, $value, $callback): void
    {
        parent::callBeforeAndAfterSyncHooks($name, $value, $callback);
    }
```

**Comment by danharrin** [comment]
@hxnk too many bad experiences with people not submitting real reproduction repositories, I flat out refuse to waste any time anymore. This isn't just about this issue, this is a wider problem. Sorry

**Comment by hxnk** [comment]
@danharrin reproduction repo is up! https://github.com/hxnk/filament-state-updated-bug

I've created a Users page with a table, running the action on a record proves the bug exists in the notification shown on submit.

Cause explained here, (and no PR because I'm not sure how you would want to solve this...):

> @danharrin I can confirm this does indeed happen when creating a Page (child of `Filament\Pages\Page`), and utilizing the `InteractsWithTable` trait on that child page.
> 
> * `InteractsWithTable` utilizes `InteractsWithForms`
> * The parent `Filament\Pages\Page` utilizes `InteractsWithForms` as well (indirectly through `HasActions`)
> 
> The issue is with the parent call to `callBeforeAndAfterSyncHooks` in the `InteractsWithForms` trait here:
> 
> https://github.com/filamentphp/filament/blob/020bf2c6187317f2fe3de1fdddf80b365b9266c9/packages/forms/src/Concerns/InteractsWithForms.php#L205C3-L212
> 
> It's effectively running twice: first from the custom page context, and then from the parent page context...

**Comment by danharrin** [comment]
Can you please test if this is an issue with v3 beta? Thanks!

**Comment by danharrin** [comment]
Should be fixed by #7901

**Comment by wq9578** [comment]
I'm running into the same problem for TextInput.

A money amount stored in cents internally shall be divided by 100 for the form.
This happens twice, so it is actually divided by 10,000.

The dump shows two fully identical debug_backtrace() outputs.

Code:
```
            ->afterStateHydrated(function (TextInput $component, $state) {
                if (! is_null($state)) {
                    dump($state);
                    dump(collect(debug_backtrace())->map(fn ($e
                    ) => $e['file'].':'.$e['line']));
                    $component->state($state / 100);
                }
            })
```
